### ISSUE TITLE
A Prisma error that covers all possible errors as mentioned in `Errors reference`

### ISSUE THREAD

**Issue by Slmii** [prisma/prisma#9082]
## Problem

In any `try/catch` that runs a prisma query you'd have to check if the `error` is an instance of a specific prisma error in order to handle different kinds of errors and get the type-safety, for example if you want the the `code` or `message` property of an error.

If you want to catch all possible prisma errors you'd have to write a seperate `if` statement for all of them, for example if you want to log the error.

```js
if (error instanceof Prisma.PrismaClientKnownRequestError) {
  log(error.code, error.message)
}


```

**Comment by matthewmueller** [comment]
I believe https://github.com/prisma/prisma/issues/5040 would solve this issue.

**Comment by Slmii** [comment]
I dont believe it would, you'd still have to make a bunch of `if` statements to cover all possible scenarios, and in each scenario i'd have to call the `log` function to have typesafety.

**Comment by niklasgrewe** [comment]
any news?

**Comment by creativiii** [comment]
Would love a better way to handle errors. Even a property on the `error` object that returns just the error itself would be fantastic. I really don't want to return:
```
Invalid `prisma.user.update()` invocation in
/home/leonardo/ironeko-fondamenta/.next/server/pages/api.js:127:32

  124   prisma
  125 }) => {
  126   try {
→ 127     return await prisma.user.update(
  An operation failed because it depends on one or more records that were required but not found. Record to update not found. 
```

To my frontend.

**Comment by system32uwu** [comment]
Crazy that you can get this highly detailed error message from the engine

```
  20 async create(data: CreateUserDto) {
→ 21   const user = await this.prismaService.user.create({
         data: {
           email: 'mateo@nfw.gg',
           password: '$2a$10$CBE/dMtwekhH9IlocK1cPOXr9g.NfI3pCmaSmLFVOYv5yvkbBdY26',
       +   name: String,
       +   surname: String,
       ?   walletAddress?: String | null,
       ?   role?: INVESTOR | AGENT
         }
       })

Argument name for data.name is missing.
Argument surname for data.surname is missing.

Note: Lines with + are required, lines with ? are optional.
```

But not this from the client to just deliver it in the response: 
```
{
  code: 123,
  meta: ['name', 'surname'],
  message: 'missing required fields'
}
```

**Comment by Prince-Rabadiya** [comment]
> Would love a better way to handle errors. Even a property on the `error` object that returns just the error itself would be fantastic. I really don't want to return:
> 
> ```
> Invalid `prisma.user.update()` invocation in
> /home/leonardo/ironeko-fondamenta/.next/server/pages/api.js:127:32
> 
>   124   prisma
>   125 }) => {
>   126   try {
> → 127     return await prisma.user.update(
>   An operation failed because it depends on one or more records that were required but not found. Record to update not found. 
> ```
> 
> To my frontend.

### can specify it like this

```
new PrismaClient({
    errorFormat: 'minimal',
  })
```

refer this for more [info](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/error-formatting)

**Comment by VitorTheGreat** [comment]
Hello everyone!

Do we have a solution for this or creating a bunch of "if" to handle all errors is still the solution we have?

Thanks!

**Comment by PhaserEditor2D** [comment]
If I user prisma with expressjs, and there is an unhandled prisma error, then the server crashes. It would be great to have a way to catch all the unhandled errors from prisma and log it.

**Comment by system32uwu** [comment]
An example for a global error handler would be much appreciated, I haven't been able to deliver this exceptions safely to the client without making the code look ugly

currently I'm doing it like this (ugly):

```
const result = await db.movie.create(...).catch((e) => {
  return await handlePrismaErrors()(req, res);
})

if (result) {
  // keep doing stuff
} else {
  // response early return, return from here as well
 return;
}
```

and then
